### PR TITLE
feat: env var obfuscation

### PR DESCRIPTION
**Pull request by brainbicycle**
This PR resolves [MOPLAT-597] 

### Description

Points to our react-native-config fork so we have obfuscated env vars. 

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.

### Potential Follow-up

- I somewhat often run into issues where Xcode fails to build with _DOT_ENV could not be found or something similar in eigen, wondering if this is something we can fix in our fork

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[MOPLAT-597]: https://artsyproduct.atlassian.net/browse/MOPLAT-597?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -75,7 +75,7 @@
     "react-nanny": "^2.15.0",
     "react-native": "0.69.6",
     "react-native-collapsible-tab-view": "^5.0.0-rc.13",
-    "react-native-config": "^1.4.11",
+    "react-native-config": "https://github.com/artsy/react-native-config",
```
Comment (damassi) [inline package.json:78]: whoa

**Review by damassi — APPROVED** [review]
Nice and easy 🎇

File: package.json (hunk 1 above)
Comment (pvinis) [inline package.json:78]:
```suggestion
    "react-native-config": "https://github.com/artsy/react-native-config.git#v1.4.11-artsy",
```
i suggest we use this, so we have a way to version this

File: ios/Podfile.lock [hunk 2]
```diff
@@ -712,10 +712,10 @@ SPEC CHECKSUMS:
   Flipper-RSocket: d9d9ade67cbecf6ac10730304bf5607266dd2541
   FlipperKit: 5f38e9e8c3975f0e91373f0b435c2539283c60d6
   fmt: ff9d55029c625d3757ed641535fd4a75fedc7ce9
-  glog: 5337263514dd6f09803962437687240c5dc39aa4
+  glog: 3d02b25ca00c2d456734d0bcff864cbc62f6ae1a
```
Comment (pvinis) [inline ios/Podfile.lock:715]: this again lol. try also a pod install repo update? chris got it earlier today too

Comment (brainbicycle) [reply ios/Podfile.lock:715]: hmm no change from that, did that solve it for Chris?

**Review by pvinis — APPROVED** [review]
thanks!

File: ios/Podfile.lock (hunk 2 above)
Comment (pvinis) [reply ios/Podfile.lock:715]: cool. lets add it then, and hopefully it stops for everyone haha

**Pull request merged**